### PR TITLE
upgrading to newest nodejs_rules

### PR DESCRIPTION
**Pull request by robin-da**
This upgrades the nodejs bazel rules to the latest version.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: deps.bzl [hunk 1]
```diff
@@ -166,19 +166,22 @@ def daml_deps():
...
         http_archive(
             name = "build_bazel_rules_nodejs",
-            urls = ["https://github.com/bazelbuild/rules_nodejs/releases/download/0.32.2/rules_nodejs-0.32.2.tar.gz"],
-            sha256 = "6d4edbf28ff6720aedf5f97f9b9a7679401bf7fca9d14a0fff80f644a99992b4",
-            patches = ["@com_github_digital_asset_daml//bazel_tools:rules_nodejs_posix_path.patch"],
-            patch_args = ["-p1"],
+            urls = ["https://github.com/bazelbuild/rules_nodejs/releases/download/1.1.0/rules_nodejs-1.1.0.tar.gz"],
+            sha256 = "c97bf38546c220fa250ff2cc052c1a9eac977c662c1fc23eda797b0ce8e70a43",
+            # patches = ["@com_github_digital_asset_daml//bazel_tools:rules_nodejs_posix_path.patch"],
+            # patch_args = ["-p1"],
```
Comment (aherrmann-da) [inline deps.bzl]: If this patch is no longer required, can we remove it?


<details>
<summary>If it is still needed, the following applies to 1.1.0</summary>

``` patch
diff --git a/internal/pkg_npm/pkg_npm.bzl b/internal/pkg_npm/pkg_npm.bzl
index be857b3f..410a90b8 100644
--- a/internal/pkg_npm/pkg_npm.bzl
+++ b/internal/pkg_npm/pkg_npm.bzl
@@ -202,6 +202,7 @@ def create_package(ctx, deps_sources, nested_packages):
     if stamp:
         inputs.append(ctx.version_file)
 
+    posix = ctx.toolchains["@rules_sh//sh/posix:toolchain_type"]
     ctx.actions.run(
         progress_message = "Assembling npm package %s" % package_dir.short_path,
         mnemonic = "AssembleNpmPackage",
@@ -209,6 +210,9 @@ def create_package(ctx, deps_sources, nested_packages):
         inputs = inputs,
         outputs = [package_dir, ctx.outputs.pack, ctx.outputs.publish],
         arguments = [args],
+        env = {
+            "PATH": ctx.host_configuration.host_path_separator.join(posix.paths),
+        },
     )
     return package_dir
 
@@ -246,4 +250,5 @@ pkg_npm = rule(
     attrs = PKG_NPM_ATTRS,
     doc = _DOC,
     outputs = PKG_NPM_OUTPUTS,
+    toolchains = ["@rules_sh//sh/posix:toolchain_type"],
 )
```

</details>

File: docs/BUILD.bazel [hunk 2]
```diff
@@ -25,7 +25,83 @@ genrule(
             "theme/da_theme/**",
         ],
     ) + [
+        "@npm//acorn",
```
Comment (aherrmann-da) [inline docs/BUILD.bazel]: Could you add a comment pointing to [this issue](https://github.com/bazelbuild/rules_nodejs/issues/1553) for future reference?

File: navigator/frontend/webpack.config.js [hunk 3]
```diff
@@ -52,8 +52,13 @@ const APP_NAME = 'Navigator';
  */
 module.exports = (env) => {
   const paths_case_check = env && env.paths_case_check  || 'true';
-  const in_dir           = env && env.bazel_in_dir  || __dirname;
-  const out_dir          = env && env.bazel_out_dir || path.join(__dirname, 'dist');
+  // const in_dir           = env && env.bazel_in_dir  || __dirname;
```
Comment (aherrmann-da) [inline navigator/frontend/webpack.config.js]: Could you add a comment pointing to [this issue](https://github.com/bazelbuild/rules_nodejs/issues/1554) for future reference?

File: docs/BUILD.bazel (hunk 2 above)
Comment (robin-da) [reply docs/BUILD.bazel]: added a comment.

File: navigator/frontend/webpack.config.js (hunk 3 above)
Comment (robin-da) [reply navigator/frontend/webpack.config.js]: added a comment.

**Pull request merged**